rust/src/rfb/rfb.rs

```diff
@@ -633,3 +633,86 @@ pub unsafe extern "C" fn rs_rfb_register_parser() {
         SCLogDebug!("Protocol detector and parser disabled for RFB.");
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_error_state() {
+        let mut state = RFBState::new();
+
+        let buf = &[
+            0x05, 0x00, 0x03, 0x20, 0x20, 0x18, 0x00, 0x01, 0x00, 0xff, 0x00, 0xff, 0x00, 0xff,
+            0x10, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x1e, 0x61, 0x6e, 0x65, 0x61,
+            0x67, 0x6c, 0x65, 0x73, 0x40, 0x6c, 0x6f, 0x63, 0x61, 0x6c, 0x68, 0x6f, 0x73, 0x74,
+            0x2e, 0x6c, 0x6f, 0x63, 0x61, 0x6c, 0x64, 0x6f, 0x6d, 0x61, 0x69, 0x6e,
+        ];
+        let r = state.parse_request(buf);
+
+        assert_eq!(
+            r,
+            AppLayerResult {
+                status: -1,
+                consumed: 0,
+                needed: 0
+            }
+        );
+    }
+    #[test]
+    fn test_request_response_events() {
+        let mut init_state = RFBState::new();
```

Comment:
I find it cool how you were able to find a valid buffer to test so many States, but I'm not sure about testing the request and response functions in the same unit test. Let's try to get feedback from some of the others about this bit :)

Reply:
> I find it cool how you were able to find a valid buffer to test so many States, but I'm not sure about testing the request and response functions in the same unit test. Let's try to get feedback from some of the others about this bit :)

The alternative would be to test either just one request or response in one test and add multiple tests to test each state.
I am testing both functions in one right now mainly because the input to one function is the output of the other so they are interlinked and thought maybe it will be good if the flow is tested. I myself felt that it might be too much for a unit test but wasn't sure about the right approach here.

Reply:
Your thinking is reasonable, in my opinion! I just don't think I can decide on that one alone...